app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/DownloadOfflineProductService.java

```diff
@@ -0,0 +1,260 @@
+import android.util.Log;
+import android.util.Pair;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import okhttp3.ResponseBody;
+import openfoodfacts.github.scrachx.openfood.R;
+import openfoodfacts.github.scrachx.openfood.network.OpenFoodAPIClient;
+import openfoodfacts.github.scrachx.openfood.views.MainActivity;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class DownloadOfflineProductService extends IntentService {
+
+    private static final String TAG = "DOPService";
+    public static final String IDENTIFIER = "INTENT_IDENTIFIER_FOR_BRAODCASTING";
+    public static final String DOWNLOAD_PROGRESS_UPDATE_KEY = "PROGRESS_UPDATE_FOR_DOWNLOADING";
+    public static boolean isDownloadOfflineProductServiceRunning = false;
+    SharedPreferences settings;
+    NotificationManager notificationManager;
+    NotificationCompat.Builder builder;
+
+    DownloadZipFileTask downloadZipFileTask;
+
+    Intent broadcastIntent;
+
+    public DownloadOfflineProductService() {
+        super("DownloadOfflineProductService");
+        broadcastIntent = new Intent(IDENTIFIER);
+    }
+
+    @Override
+    protected void onHandleIntent(@Nullable Intent intent) {
+        isDownloadOfflineProductServiceRunning = true;
+        settings = getSharedPreferences("prefs", Context.MODE_PRIVATE);
+        notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            String channelId = "channel";
+            CharSequence channelName = getString(R.string.notification_channel_name);
+            int importance = NotificationManager.IMPORTANCE_DEFAULT;
+            NotificationChannel notificationChannel = new NotificationChannel(channelId, channelName, importance);
+            notificationManager.createNotificationChannel(notificationChannel);
+        }
+        builder = new NotificationCompat.Builder(this, "channel");
+
+        builder.setContentTitle(getString(R.string.app_name))
+                .setContentText(getString(R.string.txtDownloading))
+                .setAutoCancel(true)
+                .setSmallIcon(R.mipmap.ic_launcher);
+
+        doTask();
+    }
+
+    private void doTask() {
+
+        //notify downloading is started
+        builder.setContentTitle(getString(R.string.app_name))
+                .setContentText(getString(R.string.txtDownloading))
+                .setProgress(100, 0, true)
+                .setOngoing(true)
+                .setSmallIcon(R.mipmap.ic_launcher);
+
+        notificationManager.notify(7, builder.build());
+
+        //downloading data
+        OpenFoodAPIClient client = new OpenFoodAPIClient(this);
+        String fileURL = "http://fr.openfoodfacts.org/data/offline/fr.openfoodfacts.org.products.small.zip";
```

Comment:
Can you turn that into a non-translatable string so that it can work with Open Beauty Facts and the other variants ?